pandas/core/series.py

```diff
@@ -2301,6 +2301,19 @@ def diff(self, periods=1):
         dtype: float64
         """
         result = algorithms.diff(com.values_from_object(self), periods)
+        if self.dtype in [
```

Comment:
this not a good pattern

if always passing the dtype is not working
then you can use a is_* method here

explicitly listing dtypes is a no no